sslSniffer/sslSnifferTest/snifftest.c

```diff
@@ -1290,22 +1290,30 @@ int main(int argc, char** argv)
         if (data == NULL) {
         /* grab next pcap packet */
             packetNumber++;
-            packet = pcap_next(pcap, &header);
+            if(pcap_next_ex(pcap, &header, &packet) < 0) {
+                hadBadPacket = 1;
+                break;
+            }
         }
 
         if (packet) {
-            if (header.caplen > 40)  { /* min ip(20) + min tcp(20) */
+            if (header->caplen > 40)  { /* min ip(20) + min tcp(20) */
                 packet        += frame;
-                header.caplen -= frame;
+                header->caplen -= frame;
             }
             else {
                 /* packet doesn't contain minimum ip/tcp header */
                 continue;
             }
+            int ptype = pcap_datalink(pcap);
```

Comment:
Avoid mixed-declaration. Move `int ptype` up. Also please add comment documenting what Linux SLL is and what the 2 bytes are.